src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Stream.cs

```diff
@@ -738,6 +784,12 @@ public override ValueTask WriteAsync(ReadOnlyMemory<byte> buffer, CancellationTo
                         return new ValueTask(Task.FromException(new ObjectDisposedException(nameof(Http2WriteStream))));
                     }
 
+                    // TODO: until #9071 is fixed
+                    if (http2Stream._abortException is OperationCanceledException)
+                    {
+                        throw http2Stream._abortException;
```

Comment:
```C#
ExceptionDispatchInfo.Throw(http2Stream._abortException);
```

Reply:
Why do we propagate this exception only if it's an OCE?  Why not any time it's not null?